innopoints/core/email.py

```diff
@@ -0,0 +1,118 @@
+'''Email-generating module.'''
+from flask import current_app
+from flask_mail import Message
+
+from innopoints.models import NotificationType, StockChangeStatus, ApplicationStatus
+from innopoints.schemas import PayloadSchema
+
+
+def convert_links(fragment):
+    '''Map two-element tuples to links.'''
+    if isinstance(fragment, str):
+        return fragment
+    return f'<a href="{current_app.config["FRONTEND_BASE"]}{fragment[1]}">{fragment[0]}</a>'
+
+# pylint: disable=too-many-branches
+def get_email_message(notification_type, payload, recipient_email):
+    '''Convert the payload to a email Message object based on the notification type.'''
+    header = None
+    body = None
+    payload = PayloadSchema().fill_data(payload.copy())
+
+    if notification_type == NotificationType.purchase_status_changed:
+        header = 'Purchase status changed'
+        body = ['Your ']
+        if payload['product'].type is None:
+            body.append((payload['product'].name, f"/products/{payload['product'].id}"))
```

Comment:
I think a dict would be much more readable than a tuple
```suggestion
            body.append({
                'text': payload['product'].name,
                'url': f"/products/{payload['product'].id}",
            })
```

Reply:
Nah, I don't think it's worth the overhead. It's a pretty messed up piece of code anyway